doc/iterations/2024.3/shapes/group_tools.adoc

```diff
@@ -0,0 +1,138 @@
+= Add initial support for tools working on groups of elements in diagrams
+
+== Problem
+
+The tools that can currently be defined in a diagram can only apply to one or two elements.
+
+* Most tools only apply to the element the user has selected and on which palette's the tool appears.
+* It is possible to have tools that take 2 elements as input using either the "edge creation tool" or by asking for an additional input element using a "Selection Description".
+Both these mechanisms are limited to 2 elements and each impose a very specific workflow/UI which is not adapted to many tools.
+
+== Key Result
+
+It should be possible for studio makers to specify tools that apply to groups of elements as a whole.
+As a concrete example, on a UML class diagram, the user should be able to select mutliple classes (or any kind of PackageElement) and be proposed with a tool "Wrap in new Package".
+
+== Solution
+
+The solution will involve two parts:
+
+1. First a new way to select a group of elements from the UI.
+2. Then the possibility for a specifier to define a new kind of tool which applies to such a group of elements.
+
+=== Group Selection
+
+We will add support for selecting multiple elements by (rectangular) zone.
+
+To select a zone, the user will need to:
+
+1. press the "Shift" key and while it is pressed:
+2. push the left mouse button to select the first corner of the zone;
+3. move the mouse (with the "Shift" key and left button still pressed) to the second corner;
+4. release the mouse button and then the Shift key.
+
+After this interaction, all the graphical elments on the diagram which are _completely inside_ the chosen zone are now selected.
+This includes nodes, edges, list items, border nodes.
```

Comment:
In case of list or container, isn't it difficult to manage these sub-elements (exclude them from the preconditionExpression for example)?